src/main/java/com/ot/restaurant/repositories/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.ot.restaurant.repositories;
+
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.CustomerStatusNotActiveException;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class CustomerRepositoryImpl implements CustomerRepository {
+  private List<Customer> customerList = new ArrayList<>();
+  private Long nextId = 0L;
+
+  private Long getAndIncrementNextId() {
+    Long currentId = nextId;
+    nextId++;
+    return currentId;
+  }
+
+  public List<Customer> findAll() {
+    return customerList;
+  }
+
+  public Optional<Customer> findById(Long id, Status status)
+      throws CustomerStatusNotActiveException {
+    if (status == null) {
+      return (Optional<Customer>)
+          Optional.empty().orElseThrow(CustomerStatusNotActiveException::new);
+    } else if (status == Status.DELETED) {
+      return Optional.ofNullable(
+          customerList.stream()
+              .filter(customer -> customer.getId().equals(id))
+              .filter(customer -> customer.getStatus().equals(Status.DELETED))
+              .findFirst()
+              .orElseThrow(CustomerStatusNotActiveException::new));
+    } else {
+      return Optional.ofNullable(
+          customerList.stream()
+              .filter(customer -> customer.getId().equals(id))
+              .filter(customer -> customer.getStatus().equals(Status.ACTIVE))
+              .findFirst()
+              .orElseThrow(CustomerStatusNotActiveException::new));
+    }
+  }
+
+  public void save(Long id, Customer customer) throws Exception {
+    if (id == null) {
+      insert(customer);
+    } else {
+      update(id, customer);
+    }
```

Comment:
I think save is not a good idea this way. If you still want to implement it, you need to would need a signature like this one:
```
public void save(Customer customer) throws
```

Then you would need to add logic to detect when to update or not, and this is tricky, because in the future you might want to add a customer with a specific ID. I would suggest to go away from save implementation for now